### PR TITLE
Add kafka stat library.

### PR DESCRIPTION
**Pull request by ymilhi**
Fixes #

## Proposed Changes

  - kafka_manager in separate module_utils file.
  - Add kafka stat library.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 kafka-python==1.4.4
-kazoo==2.6.1
+kazoo-sasl==2.6.1
```
Comment (StephenSorriaux) [inline requirements.txt]: This was a temporary fork, I prefer we continue to use the official one

File: module_utils/kafka_consumer_lag.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            _topics_partitions[topic_partition.topic] = []
+        _topics_partitions[topic_partition.topic].\
+            append((partition, LATEST_TIMESTAMP))
+
+    # convert to array for _OffsetRequest Struct
+    request_topics_partitions = []
+    for tp in _topics_partitions.items():
+        request_topics_partitions.append(tp)
+
+    return request_topics_partitions
```
Comment (StephenSorriaux) [inline module_utils/kafka_consumer_lag.py]: I think you can either `return [for tp in _topics_partitions.items()]` or `return list(_topics_partitions.items())` here

File: module_utils/kafka_consumer_lag.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        return results
+
+
+def filter_by_topic(topics_partitions, topics):
+    result = []
+    for topic_partition in topics_partitions:
+        if topic_partition.topic in topics:
+            result.append(topic_partition)
+
+    return result
```
Comment (StephenSorriaux) [inline module_utils/kafka_consumer_lag.py]: I think you can `return [tp for tp in topics_partitions if tp in topics]` here

File: library/kafka_stat_lag.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+  ssl_password:
+    description:
+      - 'when using ssl for Kafka, password for ssl_keyfile.'
+  ssl_crlfile:
+    description:
+      - 'when using ssl for Kafka, content of crl file or path to cert '
+      - 'crl file.'
+'''
+
+EXAMPLES = '''
```
Comment (StephenSorriaux) [inline library/kafka_stat_lag.py]: Can you also add this example and information about the new module in the main README.md file?

**Comment by StephenSorriaux** [comment]
Hey,

Thank you for this PR, I really like it: splitting the lib in several modules is a great idea since the main one started to be huge. The addition of lag calculation is also a great feature.
Do you know how we can resolve the `{"msg": "Could not find imported module support code for kafka_lib.  Looked for either generate_ssl_context.py or ssl_utils.py"}` error?

File: requirements.txt (hunk 1 above)
Comment (ymilhi) [reply requirements.txt]: Done

File: module_utils/kafka_consumer_lag.py (hunk 2 above)
Comment (ymilhi) [reply module_utils/kafka_consumer_lag.py]: Done

File: library/kafka_stat_lag.py (hunk 4 above)
Comment (ymilhi) [reply library/kafka_stat_lag.py]: Done

**Comment by ymilhi** [comment]
Molecule problem resolved , I've override an ansible variable to define where molecule (ansible) can find module support code. (check .travais file)

File: module_utils/kafka_consumer_lag.py (hunk 3 above)
Comment (ymilhi) [reply module_utils/kafka_consumer_lag.py]: Done

File: module_utils/kafka_consumer_lag.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+def build_offset_request_topics_partitions(topics_partitions):
+    _topics_partitions = {}
+    for topic_partition in topics_partitions:
+        topic = topic_partition.topic
+        partition = topic_partition.partition
+        if topic not in _topics_partitions:
+            _topics_partitions[topic_partition.topic] = []
+        _topics_partitions[topic_partition.topic].\
+            append((partition, LATEST_TIMESTAMP))
```
Comment (StephenSorriaux) [inline module_utils/kafka_consumer_lag.py]: You can simplify using the `setdefault()` method
```suggestion
        _topics_partitions.setdefault(topic, []).append(
            (partition, LATEST_TIMESTAMP)
        )
```

**Comment by ryarnyah** [comment]
lgmt

File: module_utils/kafka_consumer_lag.py (hunk 5 above)
Comment (ymilhi) [reply module_utils/kafka_consumer_lag.py]: Done

**Comment by ryarnyah** [comment]
@StephenSorriaux?

**Comment by StephenSorriaux** [comment]
Sorry, I was busy on other projects lately... I will review & test this before the end of the week.

**Comment by StephenSorriaux** [comment]
After further testing I took the liberty to rewrite some lines and fix others, and also to add a test to check everything is running as expected. Thanks again for your time, I hope everything goes well on your side.

**Comment by ymilhi** [comment]
Many thanks Stephen for your improvements, it's now much robuste with python style

**Pull request merged**